tests/unittests/Foundation/ReferenceFoundation/TestNSString.mm

```diff
@@ -595,19 +598,31 @@ guard ensureFiles([@"/tmp/ABC/temp.txt"]) else {
     {
         NSString* path = @"/tmp/";
         auto result = path.stringByDeletingLastPathComponent;
+#ifdef _WIN32
         ASSERT_OBJCEQ(result, static_cast<NSString*>(_CFGetSlashStr()));
+#else
+        ASSERT_OBJCEQ(result, @"/");
+#endif
     }
 
     {
         NSString* path = @"/tmp";
         auto result = path.stringByDeletingLastPathComponent;
+#ifdef _WIN32
         ASSERT_OBJCEQ(result, static_cast<NSString*>(_CFGetSlashStr()));
+#else
+        ASSERT_OBJCEQ(result, @"/");
+#endif
     }
 
     {
         NSString* path = @"/";
         auto result = path.stringByDeletingLastPathComponent;
+#ifdef _WIN32
         ASSERT_OBJCEQ(result, static_cast<NSString*>(_CFGetSlashStr()));
+#else
+        ASSERT_OBJCEQ(result, @"/");
+#endif
```

Comment:
1. this feels like it should be factored out.
2. This sort of change brings to light a problem in this approach. Namely we can't test *any* implementation specific details. This includes privately exported functions but also anything in the libs. UTs are currently linking with the libs to allow access to internal / private state modification (although I think security is one of the only tests to use it). What is the plan going forward to address this?